### PR TITLE
Add analytics events

### PR DESCRIPTION
**Pull request by caguilar187**
AIC-116
AIC-147
AIC-148
AIC-151
AIC-428

### CODE REVIEW

File: analytics/src/main/java/edu/artic/analytics/AnalyticsTracker.kt [hunk 1]
```diff
@@ -23,23 +27,33 @@ interface AnalyticsTracker {
...
     private val tracker = analytics.newTracker(analyticsConfig.trackingId)
 
     override fun reportEvent(category: String, action: String, label: String) {
+        val memberId = membershipPrefs.memberID
         tracker.send(HitBuilders.EventBuilder()
                 .setCategory(category)
                 .setAction(action)
+                .setCustomDimension(2, (if (memberId != null) "Member" else "None"))
+                .setCustomDimension(3, languageSelector.getAppLocale().nameOfLanguageForAnalytics())
+                .setCustomDimension(4, Locale.getDefault().toString())
```
Comment (Cliabhach) [inline analytics/src/main/java/edu/artic/analytics/AnalyticsTracker.kt]: Shouldn't this be `Locale.getDefault().toLanguageTag()` instead? I'm assuming we need to use the BCP 47 format, not the backward-compatible java String (c.f. javadoc on `Locale::toString`).

File: map/src/main/kotlin/edu/artic/map/MapFragment.kt [hunk 2]
```diff
@@ -330,8 +330,8 @@ class MapFragment : BaseViewModelFragment<MapViewModel>() {
 
         viewModel.selectedArticObject
                 .withLatestFrom(viewModel.displayMode) { selected, mapMode -> selected to mapMode }
-                .filterFlatMap({ (_, mapMode) -> mapMode is MapDisplayMode.CurrentFloor }, { (selected) -> selected })
-                .subscribeBy { selected ->
+                .filterFlatMap({(_, mapMode) -> mapMode is MapDisplayMode.CurrentFloor },{(selectedObject, mapMode) -> selectedObject to mapMode as MapDisplayMode.CurrentFloor})
+                .subscribeBy { (selected, _) ->
```
Comment (Cliabhach) [inline map/src/main/kotlin/edu/artic/map/MapFragment.kt]: Does this change have any functional side-effect?

I presume it's just to make the code more readable.

File: media/src/main/java/edu/artic/media/audio/AudioPlayerService.kt [hunk 3]
```diff
@@ -423,7 +423,7 @@ class AudioPlayerService : DaggerService(), PlayerService {
     }
 
     override fun stopPlayer() {
-        analyticsTracker.reportEvent(EventCategoryName.PlayBack, AnalyticsAction.playbackInterrupted, (currentTrack as BehaviorSubject).value?.title.orEmpty())
+        analyticsTracker.reportEvent(EventCategoryName.PlayBack, AnalyticsAction.playAudioTourStop, (currentTrack as BehaviorSubject).value?.title.orEmpty())
```
Comment (Cliabhach) [inline media/src/main/java/edu/artic/media/audio/AudioPlayerService.kt]: Hold on, isn't that the event for when you play the audio at a given tour stop? I don't recall seeing anything in the ticket about removing `playbackInterrupted`.

File: membership/build.gradle [hunk 4]
```diff
@@ -16,23 +16,9 @@ android {
...
-    implementation libs.arch_navigation_ui
-
-    api libs.design_support
-    implementation libs.glide
-    implementation project(':base')
-    implementation project(':ui')
-    implementation project(':viewmodel')
-    implementation project(':navigation')
-    implementation project(':search')
+    api project(':base')
```
Comment (Cliabhach) [inline membership/build.gradle]: It really does feel nice to see all the unnecessary direct dependencies removed.

**Comment by Cliabhach** [comment]
Oh, and I think the Circle CI build is failing because `membership/src/main/res/values-en/strings.xml` isn't in the `membership_ui` module.

File: map/src/main/kotlin/edu/artic/map/MapFragment.kt (hunk 2 above)
Comment (caguilar187) [reply map/src/main/kotlin/edu/artic/map/MapFragment.kt]: no undoing this change i forgot about that little portion my first stab at adding the map analytics was wrong

Comment (Cliabhach) [reply map/src/main/kotlin/edu/artic/map/MapFragment.kt]: Well, it does make it more readable. Feel free to add it back in a separate PR.

File: media/src/main/java/edu/artic/media/audio/AudioPlayerService.kt (hunk 3 above)
Comment (caguilar187) [reply media/src/main/java/edu/artic/media/audio/AudioPlayerService.kt]: Check above in this class `playbackInterrupted` is still called when the user switches tracks. Player stopped is indicating the user stop the playback all together

File: analytics/src/main/java/edu/artic/analytics/AnalyticsTracker.kt (hunk 1 above)
Comment (caguilar187) [reply analytics/src/main/java/edu/artic/analytics/AnalyticsTracker.kt]: which toString offers

File: media/src/main/java/edu/artic/media/audio/AudioPlayerService.kt (hunk 3 above)
Comment (Cliabhach) [reply media/src/main/java/edu/artic/media/audio/AudioPlayerService.kt]: Ah, I missed that.

It _is_ weird that the new event here has the word `tour` in it....you're sure that's the right constant?

File: analytics/src/main/java/edu/artic/analytics/AnalyticsTracker.kt (hunk 1 above)
Comment (Cliabhach) [reply analytics/src/main/java/edu/artic/analytics/AnalyticsTracker.kt]: The requirements gave `en-US` as an example...I've left a comment over there, hopefully we can sort that out in the AM.

I'll accept this PR with these two locales in either format for now.

**Comment by Cliabhach** [comment]
All right, approved. Just need to wait for Circle CI.

File: media/src/main/java/edu/artic/media/audio/AudioPlayerService.kt (hunk 3 above)
Comment (caguilar187) [reply media/src/main/java/edu/artic/media/audio/AudioPlayerService.kt]: hmm you have a point there. is there a way to know if we are in a tour currently?

Comment (Cliabhach) [reply media/src/main/java/edu/artic/media/audio/AudioPlayerService.kt]: That's tracked in the `TourProgressManager` singleton. It doesn't help that we have both two distinct concepts here:

1. 'stopping a tour'
2. 'reaching a stop on the tour'

Comment (Cliabhach) [reply media/src/main/java/edu/artic/media/audio/AudioPlayerService.kt]: Fwiw the requirements say

    Triggered when the user plays an audio track by pressing play on a tour stop.

Comment (caguilar187) [reply media/src/main/java/edu/artic/media/audio/AudioPlayerService.kt]: sigh thanks

**Comment by caguilar187** [comment]
o finally CI is  working again

**Pull request merged**